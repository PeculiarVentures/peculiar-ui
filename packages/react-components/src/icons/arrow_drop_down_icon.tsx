import React from 'react';

type TIconProps = React.HTMLAttributes<HTMLOrSVGElement>;

export const ArrowDropDownIcon: React.FC<TIconProps> = (props) => (
  <svg
    {...props}
    xmlns="http://www.w3.org/2000/svg"
    fill="none"
    viewBox="0 0 24 24"
    width="24px"
    height="24px"
  >
    <path
      d="M12.39 14.512a.5.5 0 01-.78 0l-2.96-3.7A.5.5 0 019.04 10h5.92a.5.5 0 01.39.812l-2.96 3.7z"
      fill="currentColor"
    />
  </svg>
);
