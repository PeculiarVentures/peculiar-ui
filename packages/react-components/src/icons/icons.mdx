import { Meta, IconGallery, IconItem } from '@storybook/blocks';
import {
  ArrowDropDownIcon,
  ArrowRightIcon,
  CheckCircleIcon,
  CheckIcon,
  CloseCircleIcon,
  CloseIcon,
  CloseSmallIcon,
  DotIcon,
  PlusIcon,
  WarningIcon,
} from './index';

<Meta title="Components/Icons" />

# Icons

<IconGallery>
  <IconItem name="ArrowDropDown">
    <ArrowDropDownIcon />
  </IconItem>
  <IconItem name="ArrowRight">
    <ArrowRightIcon />
  </IconItem>
  <IconItem name="CheckCircle">
    <CheckCircleIcon />
  </IconItem>
  <IconItem name="Check">
    <CheckIcon />
  </IconItem>
  <IconItem name="CloseCircle">
    <CloseCircleIcon />
  </IconItem>
  <IconItem name="Close">
    <CloseIcon />
  </IconItem>
  <IconItem name="CloseSmall">
    <CloseSmallIcon />
  </IconItem>
  <IconItem name="Dot">
    <DotIcon />
  </IconItem>
  <IconItem name="Plus">
    <PlusIcon />
  </IconItem>
  <IconItem name="Warning">
    <WarningIcon />
  </IconItem>
</IconGallery>
