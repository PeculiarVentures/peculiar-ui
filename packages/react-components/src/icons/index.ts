export { CheckIcon } from './check_icon';
export { InfoIcon } from './info_icon';
export { CloseIcon } from './close_icon';
export { CloseSmallIcon } from './close_small_icon';
export { ArrowDropDownIcon } from './arrow_drop_down_icon';
