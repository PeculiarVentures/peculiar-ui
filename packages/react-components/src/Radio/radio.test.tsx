import React from 'react';
import {
  renderWithWrapper as render,
  screen,
  userEvent,
  fireEvent,
} from '../test-utils';
import { Radio } from '../index';
import { PlusIcon } from '../icons';

describe('<Radio />', () => {
  describe('Radio render variants', () => {
    it('should render as default', () => {
      render(<Radio />);

      const input = screen.getByRole('radio');

      expect(input).toBeInTheDocument();
      expect(input.tagName).toBe('INPUT');
      expect(input.getAttribute('type')).toBe('radio');
      expect(input.getAttribute('class')).toMatch(/Radio-input/i);
      expect(input).not.toBeChecked();

      const label = input.closest('label');

      expect(label).toBeInTheDocument();
      expect(label.getAttribute('for')).toEqual(input.id);
    });

    it('should have id', () => {
      render(<Radio id="test-id" />);

      const input = screen.getByRole('radio');

      expect(input.getAttribute('id')).toBe('test-id');

      const label = input.closest('label');

      expect(label).toBeInTheDocument();
      expect(label.getAttribute('for')).toEqual(input.id);
    });

    it('should be disabled', () => {
      render(<Radio disabled />);

      const input = screen.getByRole('radio');

      expect(input).toHaveAttribute('disabled');
    });

    it('should have required', () => {
      render(<Radio required />);

      const input = screen.getByRole('radio');

      expect(input).toHaveAttribute('required');
    });

    it('should have name attr', () => {
      render(<Radio name="test-name" />);

      const input = screen.getByRole('radio');

      expect(input.getAttribute('name')).toEqual('test-name');
    });

    it('should have test id', () => {
      render(<Radio data-testid="test-id" />);

      const input = screen.getByRole('radio');

      const label = input.closest('label');

      expect(label.getAttribute('data-testid')).toMatch(/test-id/i);
    });

    it('should have className', () => {
      render(<Radio className="test-cls" />);

      const input = screen.getByRole('radio');

      const label = input.closest('label');

      expect(label.getAttribute('class')).toMatch(/test-cls/i);
    });

    // TODO: do not passed, should fix in component
    // it('should have input className', () => {
    //   render(<Radio inputProps={{ className: 'test-cls' }} />);

    //   const input = screen.getByRole('radio');

    //   expect(input.getAttribute('class')).toMatch(/test-cls/i);
    // });

    it('should have custom checked icon', () => {
      render(<Radio checkedIcon={<PlusIcon data-testid="icon" />} />);

      const icon = screen.getByTestId('icon');

      expect(icon).toBeInTheDocument();
      expect(icon.tagName).toBe('svg');
      expect(icon.getAttribute('class')).toMatch(/Radio-icon/i);
    });

    it('should be checked, defaultChecked prop', () => {
      render(<Radio defaultChecked />);

      const input = screen.getByRole('radio');

      expect(input).toBeChecked();
    });

    it('should be unchecked, defaultChecked prop', () => {
      render(<Radio defaultChecked={false} />);

      const input = screen.getByRole('radio');

      expect(input).not.toBeChecked();
    });

    it('should be checked', () => {
      const onChange = jest.fn();

      render(<Radio checked onChange={onChange} />);

      const input = screen.getByRole('radio');

      expect(input).toBeChecked();
    });

    it('should be unchecked', () => {
      const onChange = jest.fn();

      render(<Radio checked={false} onChange={onChange} />);

      const input = screen.getByRole('radio');

      expect(input).not.toBeChecked();
    });

    it('should forwards ref to label element', () => {
      const ref = React.createRef<HTMLLabelElement>();

      render(<Radio ref={ref} />);

      const input = screen.getByRole('radio');

      expect(input.closest('label')).toEqual(ref.current);
    });
  });

  describe('Checkbox focus behaviour', () => {
    it('should have focus', async () => {
      render(<Radio />);

      const input = screen.getByRole('radio');

      expect(document.activeElement).not.toEqual(input);
      await userEvent.tab();
      expect(document.activeElement).toEqual(input);
      expect(input).toHaveFocus();
    });

    it("shouldn't have focus when disabled has been passed to the component", async () => {
      render(<Radio disabled />);

      const input = screen.getByRole('radio');

      expect(document.activeElement).not.toEqual(input);
      await userEvent.tab();
      expect(document.activeElement).not.toEqual(input);
      expect(input).not.toHaveFocus();
    });
  });

  describe('Checkbox keyboard behaviour', () => {
    it('should be checked after pressed on space', async () => {
      render(<Radio />);

      const input = screen.getByRole('radio');

      await userEvent.type(input, '{space}');
      expect(input).toBeChecked();
    });
  });

  describe('Checkbox click behaviour', () => {
    it('should be checked when clicked', () => {
      render(<Radio />);

      const input = screen.getByRole('radio');

      expect(input).not.toBeChecked();
      fireEvent.click(input);
      expect(input).toBeChecked();
    });

    it('should be checked when clicked and defaultChecked has been passed to the component', () => {
      render(<Radio defaultChecked />);

      const input = screen.getByRole('radio');

      expect(input).toBeChecked();
      fireEvent.click(input);
      expect(input).toBeChecked();
    });

    it('should calls onClick prop when clicked', () => {
      const handleClick = jest.fn();

      render(<Radio onClick={handleClick} />);
      fireEvent.click(screen.getByRole('radio'));
      expect(handleClick).toHaveBeenCalledTimes(1);
    });

    it('should calls onChange prop when clicked', () => {
      const onChange = jest.fn();

      render(<Radio onChange={onChange} />);
      const input = screen.getByRole('radio');

      expect(onChange).toBeCalledTimes(0);
      fireEvent.click(input);
      expect(onChange).toBeCalledTimes(1);
    });

    it('should be checked the right checkbox when clicked them in turn', () => {
      render(
        <>
          <Radio name="test-name" />
          <Radio name="test-name" />
        </>,
      );
      const [input1, input2] = screen.getAllByRole('radio');

      expect(input1).not.toBeChecked();
      expect(input2).not.toBeChecked();
      fireEvent.click(input1);
      expect(input1).toBeChecked();
      expect(input2).not.toBeChecked();
      fireEvent.click(input2);
      expect(input1).not.toBeChecked();
      expect(input2).toBeChecked();
    });
  });
});
