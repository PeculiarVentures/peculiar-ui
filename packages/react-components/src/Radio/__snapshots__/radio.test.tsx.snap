// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Radio /> Radio render colors color "primary" 1`] = `
<DocumentFragment>
  .emotion-0 {
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  width: var(--pv-size-base-4);
  height: var(--pv-size-base-4);
  border-radius: 50%;
  position: relative;
  color: var(--pv-color-gray-6);
}

.emotion-1 {
  cursor: inherit;
  width: 100%;
  height: 100%;
  margin: 0;
  padding: 0;
  outline: 0;
  border-radius: 50%;
  border-width: 2px;
  border-style: solid;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  border-color: currentColor;
  background-color: transparent;
  color: var(--pv-color-gray-9);
}

.emotion-1:before {
  top: -7px;
  left: -7px;
  right: -7px;
  bottom: -7px;
  content: "";
  position: absolute;
  border-radius: 50%;
  opacity: 0;
  background-color: currentColor;
  -webkit-transition: opacity 200ms;
  transition: opacity 200ms;
}

.emotion-1:checked+[aria-hidden] {
  color: var(--pv-color-primary);
  opacity: 1;
}

.emotion-1:not(:disabled) {
  cursor: pointer;
}

.emotion-1:not(:disabled):checked {
  color: var(--pv-color-primary-shade-1);
}

.emotion-1:not(:disabled):hover:before {
  opacity: 0.18;
}

.emotion-1:not(:disabled):focus-visible:before {
  opacity: 0.23;
}

.emotion-1:not(:disabled):active:before {
  opacity: 0.3;
}

.emotion-1:disabled {
  cursor: not-allowed;
  color: var(--pv-color-gray-6);
}

.emotion-1:disabled+[aria-hidden] {
  color: inherit;
}

.emotion-1:disabled:checked {
  color: var(--pv-color-gray-7);
}

.emotion-2 {
  display: block;
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  opacity: 0;
  pointer-events: none;
}

<label
    class="emotion-0"
    for="test-id"
  >
    <input
      class="emotion-1"
      id="test-id"
      type="radio"
    />
    <svg
      aria-hidden="true"
      class="emotion-2"
      fill="none"
      height="24px"
      viewBox="0 0 24 24"
      width="24px"
      xmlns="http://www.w3.org/2000/svg"
    >
      <circle
        cx="12"
        cy="12"
        fill="currentColor"
        r="6"
      />
    </svg>
  </label>
</DocumentFragment>
`;

exports[`<Radio /> Radio render colors color "secondary" 1`] = `
<DocumentFragment>
  .emotion-0 {
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  width: var(--pv-size-base-4);
  height: var(--pv-size-base-4);
  border-radius: 50%;
  position: relative;
  color: var(--pv-color-gray-6);
}

.emotion-1 {
  cursor: inherit;
  width: 100%;
  height: 100%;
  margin: 0;
  padding: 0;
  outline: 0;
  border-radius: 50%;
  border-width: 2px;
  border-style: solid;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  border-color: currentColor;
  background-color: transparent;
  color: var(--pv-color-gray-9);
}

.emotion-1:before {
  top: -7px;
  left: -7px;
  right: -7px;
  bottom: -7px;
  content: "";
  position: absolute;
  border-radius: 50%;
  opacity: 0;
  background-color: currentColor;
  -webkit-transition: opacity 200ms;
  transition: opacity 200ms;
}

.emotion-1:checked+[aria-hidden] {
  color: var(--pv-color-secondary);
  opacity: 1;
}

.emotion-1:not(:disabled) {
  cursor: pointer;
}

.emotion-1:not(:disabled):checked {
  color: var(--pv-color-secondary-shade-1);
}

.emotion-1:not(:disabled):hover:before {
  opacity: 0.18;
}

.emotion-1:not(:disabled):focus-visible:before {
  opacity: 0.23;
}

.emotion-1:not(:disabled):active:before {
  opacity: 0.3;
}

.emotion-1:disabled {
  cursor: not-allowed;
  color: var(--pv-color-gray-6);
}

.emotion-1:disabled+[aria-hidden] {
  color: inherit;
}

.emotion-1:disabled:checked {
  color: var(--pv-color-gray-7);
}

.emotion-2 {
  display: block;
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  opacity: 0;
  pointer-events: none;
}

<label
    class="emotion-0"
    for="test-id"
  >
    <input
      class="emotion-1"
      id="test-id"
      type="radio"
    />
    <svg
      aria-hidden="true"
      class="emotion-2"
      fill="none"
      height="24px"
      viewBox="0 0 24 24"
      width="24px"
      xmlns="http://www.w3.org/2000/svg"
    >
      <circle
        cx="12"
        cy="12"
        fill="currentColor"
        r="6"
      />
    </svg>
  </label>
</DocumentFragment>
`;

exports[`<Radio /> Radio render variants should be checked, defaultChecked prop 1`] = `
<DocumentFragment>
  .emotion-0 {
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  width: var(--pv-size-base-4);
  height: var(--pv-size-base-4);
  border-radius: 50%;
  position: relative;
  color: var(--pv-color-gray-6);
}

.emotion-1 {
  cursor: inherit;
  width: 100%;
  height: 100%;
  margin: 0;
  padding: 0;
  outline: 0;
  border-radius: 50%;
  border-width: 2px;
  border-style: solid;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  border-color: currentColor;
  background-color: transparent;
  color: var(--pv-color-gray-9);
}

.emotion-1:before {
  top: -7px;
  left: -7px;
  right: -7px;
  bottom: -7px;
  content: "";
  position: absolute;
  border-radius: 50%;
  opacity: 0;
  background-color: currentColor;
  -webkit-transition: opacity 200ms;
  transition: opacity 200ms;
}

.emotion-1:checked+[aria-hidden] {
  color: var(--pv-color-primary);
  opacity: 1;
}

.emotion-1:not(:disabled) {
  cursor: pointer;
}

.emotion-1:not(:disabled):checked {
  color: var(--pv-color-primary-shade-1);
}

.emotion-1:not(:disabled):hover:before {
  opacity: 0.18;
}

.emotion-1:not(:disabled):focus-visible:before {
  opacity: 0.23;
}

.emotion-1:not(:disabled):active:before {
  opacity: 0.3;
}

.emotion-1:disabled {
  cursor: not-allowed;
  color: var(--pv-color-gray-6);
}

.emotion-1:disabled+[aria-hidden] {
  color: inherit;
}

.emotion-1:disabled:checked {
  color: var(--pv-color-gray-7);
}

.emotion-2 {
  display: block;
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  opacity: 0;
  pointer-events: none;
}

<label
    class="emotion-0"
    for="test-id"
  >
    <input
      checked=""
      class="emotion-1"
      id="test-id"
      type="radio"
    />
    <svg
      aria-hidden="true"
      class="emotion-2"
      fill="none"
      height="24px"
      viewBox="0 0 24 24"
      width="24px"
      xmlns="http://www.w3.org/2000/svg"
    >
      <circle
        cx="12"
        cy="12"
        fill="currentColor"
        r="6"
      />
    </svg>
  </label>
</DocumentFragment>
`;

exports[`<Radio /> Radio render variants should be disabled 1`] = `
<DocumentFragment>
  .emotion-0 {
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  width: var(--pv-size-base-4);
  height: var(--pv-size-base-4);
  border-radius: 50%;
  position: relative;
  color: var(--pv-color-gray-6);
}

.emotion-1 {
  cursor: inherit;
  width: 100%;
  height: 100%;
  margin: 0;
  padding: 0;
  outline: 0;
  border-radius: 50%;
  border-width: 2px;
  border-style: solid;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  border-color: currentColor;
  background-color: transparent;
  color: var(--pv-color-gray-9);
}

.emotion-1:before {
  top: -7px;
  left: -7px;
  right: -7px;
  bottom: -7px;
  content: "";
  position: absolute;
  border-radius: 50%;
  opacity: 0;
  background-color: currentColor;
  -webkit-transition: opacity 200ms;
  transition: opacity 200ms;
}

.emotion-1:checked+[aria-hidden] {
  color: var(--pv-color-primary);
  opacity: 1;
}

.emotion-1:not(:disabled) {
  cursor: pointer;
}

.emotion-1:not(:disabled):checked {
  color: var(--pv-color-primary-shade-1);
}

.emotion-1:not(:disabled):hover:before {
  opacity: 0.18;
}

.emotion-1:not(:disabled):focus-visible:before {
  opacity: 0.23;
}

.emotion-1:not(:disabled):active:before {
  opacity: 0.3;
}

.emotion-1:disabled {
  cursor: not-allowed;
  color: var(--pv-color-gray-6);
}

.emotion-1:disabled+[aria-hidden] {
  color: inherit;
}

.emotion-1:disabled:checked {
  color: var(--pv-color-gray-7);
}

.emotion-2 {
  display: block;
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  opacity: 0;
  pointer-events: none;
}

<label
    class="emotion-0"
    for="test-id"
  >
    <input
      class="emotion-1"
      disabled=""
      id="test-id"
      type="radio"
    />
    <svg
      aria-hidden="true"
      class="emotion-2"
      fill="none"
      height="24px"
      viewBox="0 0 24 24"
      width="24px"
      xmlns="http://www.w3.org/2000/svg"
    >
      <circle
        cx="12"
        cy="12"
        fill="currentColor"
        r="6"
      />
    </svg>
  </label>
</DocumentFragment>
`;

exports[`<Radio /> Radio render variants should have className 1`] = `
<DocumentFragment>
  .emotion-0 {
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  width: var(--pv-size-base-4);
  height: var(--pv-size-base-4);
  border-radius: 50%;
  position: relative;
  color: var(--pv-color-gray-6);
}

.emotion-1 {
  cursor: inherit;
  width: 100%;
  height: 100%;
  margin: 0;
  padding: 0;
  outline: 0;
  border-radius: 50%;
  border-width: 2px;
  border-style: solid;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  border-color: currentColor;
  background-color: transparent;
  color: var(--pv-color-gray-9);
}

.emotion-1:before {
  top: -7px;
  left: -7px;
  right: -7px;
  bottom: -7px;
  content: "";
  position: absolute;
  border-radius: 50%;
  opacity: 0;
  background-color: currentColor;
  -webkit-transition: opacity 200ms;
  transition: opacity 200ms;
}

.emotion-1:checked+[aria-hidden] {
  color: var(--pv-color-primary);
  opacity: 1;
}

.emotion-1:not(:disabled) {
  cursor: pointer;
}

.emotion-1:not(:disabled):checked {
  color: var(--pv-color-primary-shade-1);
}

.emotion-1:not(:disabled):hover:before {
  opacity: 0.18;
}

.emotion-1:not(:disabled):focus-visible:before {
  opacity: 0.23;
}

.emotion-1:not(:disabled):active:before {
  opacity: 0.3;
}

.emotion-1:disabled {
  cursor: not-allowed;
  color: var(--pv-color-gray-6);
}

.emotion-1:disabled+[aria-hidden] {
  color: inherit;
}

.emotion-1:disabled:checked {
  color: var(--pv-color-gray-7);
}

.emotion-2 {
  display: block;
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  opacity: 0;
  pointer-events: none;
}

<label
    class="test-cls emotion-0"
    for="test-id"
  >
    <input
      class="emotion-1"
      id="test-id"
      type="radio"
    />
    <svg
      aria-hidden="true"
      class="emotion-2"
      fill="none"
      height="24px"
      viewBox="0 0 24 24"
      width="24px"
      xmlns="http://www.w3.org/2000/svg"
    >
      <circle
        cx="12"
        cy="12"
        fill="currentColor"
        r="6"
      />
    </svg>
  </label>
</DocumentFragment>
`;

exports[`<Radio /> Radio render variants should have custom checked icon 1`] = `
<DocumentFragment>
  .emotion-0 {
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  width: var(--pv-size-base-4);
  height: var(--pv-size-base-4);
  border-radius: 50%;
  position: relative;
  color: var(--pv-color-gray-6);
}

.emotion-1 {
  cursor: inherit;
  width: 100%;
  height: 100%;
  margin: 0;
  padding: 0;
  outline: 0;
  border-radius: 50%;
  border-width: 2px;
  border-style: solid;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  border-color: currentColor;
  background-color: transparent;
  color: var(--pv-color-gray-9);
}

.emotion-1:before {
  top: -7px;
  left: -7px;
  right: -7px;
  bottom: -7px;
  content: "";
  position: absolute;
  border-radius: 50%;
  opacity: 0;
  background-color: currentColor;
  -webkit-transition: opacity 200ms;
  transition: opacity 200ms;
}

.emotion-1:checked+[aria-hidden] {
  color: var(--pv-color-primary);
  opacity: 1;
}

.emotion-1:not(:disabled) {
  cursor: pointer;
}

.emotion-1:not(:disabled):checked {
  color: var(--pv-color-primary-shade-1);
}

.emotion-1:not(:disabled):hover:before {
  opacity: 0.18;
}

.emotion-1:not(:disabled):focus-visible:before {
  opacity: 0.23;
}

.emotion-1:not(:disabled):active:before {
  opacity: 0.3;
}

.emotion-1:disabled {
  cursor: not-allowed;
  color: var(--pv-color-gray-6);
}

.emotion-1:disabled+[aria-hidden] {
  color: inherit;
}

.emotion-1:disabled:checked {
  color: var(--pv-color-gray-7);
}

.emotion-2 {
  display: block;
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  opacity: 0;
  pointer-events: none;
}

<label
    class="emotion-0"
    for="test-id"
  >
    <input
      class="emotion-1"
      id="test-id"
      type="radio"
    />
    <svg
      aria-hidden="true"
      class="emotion-2"
      fill="none"
      height="24px"
      viewBox="0 0 24 24"
      width="24px"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M6 12.5h12M12 18.5v-12"
        stroke="currentColor"
        stroke-width="1.5"
      />
    </svg>
  </label>
</DocumentFragment>
`;

exports[`<Radio /> Radio render variants should have name attr 1`] = `
<DocumentFragment>
  .emotion-0 {
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  width: var(--pv-size-base-4);
  height: var(--pv-size-base-4);
  border-radius: 50%;
  position: relative;
  color: var(--pv-color-gray-6);
}

.emotion-1 {
  cursor: inherit;
  width: 100%;
  height: 100%;
  margin: 0;
  padding: 0;
  outline: 0;
  border-radius: 50%;
  border-width: 2px;
  border-style: solid;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  border-color: currentColor;
  background-color: transparent;
  color: var(--pv-color-gray-9);
}

.emotion-1:before {
  top: -7px;
  left: -7px;
  right: -7px;
  bottom: -7px;
  content: "";
  position: absolute;
  border-radius: 50%;
  opacity: 0;
  background-color: currentColor;
  -webkit-transition: opacity 200ms;
  transition: opacity 200ms;
}

.emotion-1:checked+[aria-hidden] {
  color: var(--pv-color-primary);
  opacity: 1;
}

.emotion-1:not(:disabled) {
  cursor: pointer;
}

.emotion-1:not(:disabled):checked {
  color: var(--pv-color-primary-shade-1);
}

.emotion-1:not(:disabled):hover:before {
  opacity: 0.18;
}

.emotion-1:not(:disabled):focus-visible:before {
  opacity: 0.23;
}

.emotion-1:not(:disabled):active:before {
  opacity: 0.3;
}

.emotion-1:disabled {
  cursor: not-allowed;
  color: var(--pv-color-gray-6);
}

.emotion-1:disabled+[aria-hidden] {
  color: inherit;
}

.emotion-1:disabled:checked {
  color: var(--pv-color-gray-7);
}

.emotion-2 {
  display: block;
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  opacity: 0;
  pointer-events: none;
}

<label
    class="emotion-0"
    for="test-id"
  >
    <input
      class="emotion-1"
      id="test-id"
      name="test-name"
      type="radio"
    />
    <svg
      aria-hidden="true"
      class="emotion-2"
      fill="none"
      height="24px"
      viewBox="0 0 24 24"
      width="24px"
      xmlns="http://www.w3.org/2000/svg"
    >
      <circle
        cx="12"
        cy="12"
        fill="currentColor"
        r="6"
      />
    </svg>
  </label>
</DocumentFragment>
`;

exports[`<Radio /> Radio render variants should have required 1`] = `
<DocumentFragment>
  .emotion-0 {
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  width: var(--pv-size-base-4);
  height: var(--pv-size-base-4);
  border-radius: 50%;
  position: relative;
  color: var(--pv-color-gray-6);
}

.emotion-1 {
  cursor: inherit;
  width: 100%;
  height: 100%;
  margin: 0;
  padding: 0;
  outline: 0;
  border-radius: 50%;
  border-width: 2px;
  border-style: solid;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  border-color: currentColor;
  background-color: transparent;
  color: var(--pv-color-gray-9);
}

.emotion-1:before {
  top: -7px;
  left: -7px;
  right: -7px;
  bottom: -7px;
  content: "";
  position: absolute;
  border-radius: 50%;
  opacity: 0;
  background-color: currentColor;
  -webkit-transition: opacity 200ms;
  transition: opacity 200ms;
}

.emotion-1:checked+[aria-hidden] {
  color: var(--pv-color-primary);
  opacity: 1;
}

.emotion-1:not(:disabled) {
  cursor: pointer;
}

.emotion-1:not(:disabled):checked {
  color: var(--pv-color-primary-shade-1);
}

.emotion-1:not(:disabled):hover:before {
  opacity: 0.18;
}

.emotion-1:not(:disabled):focus-visible:before {
  opacity: 0.23;
}

.emotion-1:not(:disabled):active:before {
  opacity: 0.3;
}

.emotion-1:disabled {
  cursor: not-allowed;
  color: var(--pv-color-gray-6);
}

.emotion-1:disabled+[aria-hidden] {
  color: inherit;
}

.emotion-1:disabled:checked {
  color: var(--pv-color-gray-7);
}

.emotion-2 {
  display: block;
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  opacity: 0;
  pointer-events: none;
}

<label
    class="emotion-0"
    for="test-id"
  >
    <input
      class="emotion-1"
      id="test-id"
      required=""
      type="radio"
    />
    <svg
      aria-hidden="true"
      class="emotion-2"
      fill="none"
      height="24px"
      viewBox="0 0 24 24"
      width="24px"
      xmlns="http://www.w3.org/2000/svg"
    >
      <circle
        cx="12"
        cy="12"
        fill="currentColor"
        r="6"
      />
    </svg>
  </label>
</DocumentFragment>
`;

exports[`<Radio /> Radio render variants should have test id 1`] = `
<DocumentFragment>
  .emotion-0 {
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  width: var(--pv-size-base-4);
  height: var(--pv-size-base-4);
  border-radius: 50%;
  position: relative;
  color: var(--pv-color-gray-6);
}

.emotion-1 {
  cursor: inherit;
  width: 100%;
  height: 100%;
  margin: 0;
  padding: 0;
  outline: 0;
  border-radius: 50%;
  border-width: 2px;
  border-style: solid;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  border-color: currentColor;
  background-color: transparent;
  color: var(--pv-color-gray-9);
}

.emotion-1:before {
  top: -7px;
  left: -7px;
  right: -7px;
  bottom: -7px;
  content: "";
  position: absolute;
  border-radius: 50%;
  opacity: 0;
  background-color: currentColor;
  -webkit-transition: opacity 200ms;
  transition: opacity 200ms;
}

.emotion-1:checked+[aria-hidden] {
  color: var(--pv-color-primary);
  opacity: 1;
}

.emotion-1:not(:disabled) {
  cursor: pointer;
}

.emotion-1:not(:disabled):checked {
  color: var(--pv-color-primary-shade-1);
}

.emotion-1:not(:disabled):hover:before {
  opacity: 0.18;
}

.emotion-1:not(:disabled):focus-visible:before {
  opacity: 0.23;
}

.emotion-1:not(:disabled):active:before {
  opacity: 0.3;
}

.emotion-1:disabled {
  cursor: not-allowed;
  color: var(--pv-color-gray-6);
}

.emotion-1:disabled+[aria-hidden] {
  color: inherit;
}

.emotion-1:disabled:checked {
  color: var(--pv-color-gray-7);
}

.emotion-2 {
  display: block;
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  opacity: 0;
  pointer-events: none;
}

<label
    class="emotion-0"
    data-testid="test-id"
    for="test-id"
  >
    <input
      class="emotion-1"
      id="test-id"
      type="radio"
    />
    <svg
      aria-hidden="true"
      class="emotion-2"
      fill="none"
      height="24px"
      viewBox="0 0 24 24"
      width="24px"
      xmlns="http://www.w3.org/2000/svg"
    >
      <circle
        cx="12"
        cy="12"
        fill="currentColor"
        r="6"
      />
    </svg>
  </label>
</DocumentFragment>
`;

exports[`<Radio /> Radio render variants should render as default 1`] = `
<DocumentFragment>
  .emotion-0 {
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  width: var(--pv-size-base-4);
  height: var(--pv-size-base-4);
  border-radius: 50%;
  position: relative;
  color: var(--pv-color-gray-6);
}

.emotion-1 {
  cursor: inherit;
  width: 100%;
  height: 100%;
  margin: 0;
  padding: 0;
  outline: 0;
  border-radius: 50%;
  border-width: 2px;
  border-style: solid;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  border-color: currentColor;
  background-color: transparent;
  color: var(--pv-color-gray-9);
}

.emotion-1:before {
  top: -7px;
  left: -7px;
  right: -7px;
  bottom: -7px;
  content: "";
  position: absolute;
  border-radius: 50%;
  opacity: 0;
  background-color: currentColor;
  -webkit-transition: opacity 200ms;
  transition: opacity 200ms;
}

.emotion-1:checked+[aria-hidden] {
  color: var(--pv-color-primary);
  opacity: 1;
}

.emotion-1:not(:disabled) {
  cursor: pointer;
}

.emotion-1:not(:disabled):checked {
  color: var(--pv-color-primary-shade-1);
}

.emotion-1:not(:disabled):hover:before {
  opacity: 0.18;
}

.emotion-1:not(:disabled):focus-visible:before {
  opacity: 0.23;
}

.emotion-1:not(:disabled):active:before {
  opacity: 0.3;
}

.emotion-1:disabled {
  cursor: not-allowed;
  color: var(--pv-color-gray-6);
}

.emotion-1:disabled+[aria-hidden] {
  color: inherit;
}

.emotion-1:disabled:checked {
  color: var(--pv-color-gray-7);
}

.emotion-2 {
  display: block;
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  opacity: 0;
  pointer-events: none;
}

<label
    class="emotion-0"
    for="test-id"
  >
    <input
      class="emotion-1"
      id="test-id"
      type="radio"
    />
    <svg
      aria-hidden="true"
      class="emotion-2"
      fill="none"
      height="24px"
      viewBox="0 0 24 24"
      width="24px"
      xmlns="http://www.w3.org/2000/svg"
    >
      <circle
        cx="12"
        cy="12"
        fill="currentColor"
        r="6"
      />
    </svg>
  </label>
</DocumentFragment>
`;
